From: Lucas Kanashiro <lucas.kanashiro@canonical.com>
Date: Mon, 13 Sep 2021 18:27:58 -0300
Subject: Make resolv accept dash in zone_id of IPv6 link local addresses

One could use the network interface name as zone_id and it might contain
a dash in it.

Fixes [Bug #18165]

Bug: https://bugs.ruby-lang.org/issues/18165
Forwarded: yes, https://bugs.ruby-lang.org/issues/18165
---
 lib/resolv.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/resolv.rb b/lib/resolv.rb
index b69c704..5de6a1f 100644
--- a/lib/resolv.rb
+++ b/lib/resolv.rb
@@ -2464,7 +2464,7 @@ class Resolv
     Regex_8HexLinkLocal = /\A
       [Ff][Ee]80
       (?::[0-9A-Fa-f]{1,4}){7}
-      %[0-9A-Za-z]+
+      %[0-9A-Za-z\-]+
       \z/x
 
     ##
@@ -2478,7 +2478,7 @@ class Resolv
         |
         :((?:[0-9A-Fa-f]{1,4}(?::[0-9A-Fa-f]{1,4})*)?)
       )?
-      :[0-9A-Fa-f]{1,4}%[0-9A-Za-z.]+
+      :[0-9A-Fa-f]{1,4}%[0-9A-Za-z.\-]+
       \z/x
 
     ##
